politeiad/cache/cockroachdb/decred.go

```diff
@@ -884,6 +884,243 @@ func (d *decred) cmdTokenInventory(payload string) (string, error) {
+	}
+
+	for _, av := range avs {
+		avMap[av.Token] = av
+	}
+
+	return avMap, nil
+}
+
+// getStartVotesForAuthorizeVotes looks up the start votes for records which
+// have been authorized to start voting.
+func (d *decred) getStartVotesForAuthorizeVotes(avMap map[string]AuthorizeVote) (map[string]StartVote, error) {
+	svMap := make(map[string]StartVote)
+
+	if len(avMap) == 0 {
+		return svMap, nil
+	}
+
+	tokens := make([]string, 0, len(avMap))
+	for token := range avMap {
+		tokens = append(tokens, token)
+	}
+
+	svs := make([]StartVote, 0, len(tokens))
+	err := d.recordsdb.
+		Where("token IN (?)", tokens).
+		Preload("Options").
+		Find(&svs).
+		Error
+
+	if err != nil {
+		return nil, err
+	}
+	for _, sv := range svs {
+		svMap[sv.Token] = sv
+	}
+
+	return svMap, nil
+}
+
+// getVoteResultsForStartVotes retrieves vote results for records that have
+// begun the voting process. Results are lazily loaded into this table, so
+// some results will need to be manually looked up in the CastVote table.
+func (d *decred) getVoteResultsForStartVotes(svMap map[string]StartVote) (map[string][]decredplugin.VoteOptionResult, error) {
+
+	resMap := make(map[string][]decredplugin.VoteOptionResult)
+
+	if len(svMap) == 0 {
+		return resMap, nil
+	}
+
+	tokens := make([]string, 0, len(svMap))
+	for token := range svMap {
+		tokens = append(tokens, token)
+	}
+
+	vrs := make([]VoteResults, 0, len(tokens))
+	err := d.recordsdb.
+		Where("token IN (?)", tokens).
+		Preload("Results").
+		Preload("Results.Option").
+		Find(&vrs).
+		Error
+	if err != nil {
+		return nil, err
+	}
+
+	for _, vr := range vrs {
+		resMap[vr.Token] = convertVoteOptionResultsToDecred(vr.Results)
+	}
+
+	return resMap, nil
+}
+
+// lookupResultsForVoteOptions looks in the CastVote to see how many votes
+// each option has received.
+func (d *decred) lookupResultsForVoteOptions(options []VoteOption) ([]decredplugin.VoteOptionResult, error) {
+
+	results := make([]decredplugin.VoteOptionResult, 0, len(options))
+
+	for _, v := range options {
+		var votes uint64
+		tokenVoteBit := v.Token + strconv.FormatUint(v.Bits, 16)
+		err := d.recordsdb.
+			Model(&CastVote{}).
+			Where("token_vote_bit = ?", tokenVoteBit).
+			Count(&votes).
+			Error
+		if err != nil {
+			return nil, err
+		}
+
+		results = append(results,
+			decredplugin.VoteOptionResult{
+				ID:          v.ID,
+				Description: v.Description,
+				Bits:        v.Bits,
+				Votes:       votes,
+			})
+	}
+
+	return results, nil
+}
+
+// manuallyLookupNewVoteResults manually looks up vote results for records
+// for which voting has started but there is no vote option result in resMap.
+// This is meant to be called after resMap has been populated with the records
+// for which the vote results have been lazily loaded into the VoteResults
+// table.
+func (d *decred) manuallyLookupNewVoteResults(resMap map[string][]decredplugin.VoteOptionResult, svMap map[string]StartVote) error {
+
+	for token, sv := range svMap {
```

Comment:
Wouldn't it be simpler to have this code in the `getVoteResults` function instead of breaking it out into its own function?  `getVoteResults` only returning the results for certain proposals seems like unnecessary complexity to me.